File: src/main/java/me/zeroeightsix/kami/command/commands/EntityStatsCommand.java
package me.zeroeightsix.kami.command.commands;

import me.zeroeightsix.kami.command.Command;
import me.zeroeightsix.kami.manager.managers.UUIDManager;
import net.minecraft.entity.EntityLivingBase;
import net.minecraft.entity.passive.AbstractHorse;
import org.kamiblue.capeapi.PlayerProfile;
import org.kamiblue.commons.utils.MathUtils;

import static me.zeroeightsix.kami.util.text.MessageSendHelper.sendChatMessage;
import static me.zeroeightsix.kami.util.text.MessageSendHelper.sendErrorMessage;

/**
 * Created by d1gress/Qther on 25/11/2017, updated on 16/12/2019
 * Updated by EmotionalLove on 16/12/2019
 * Updated by Dewy on 4th April, 2020
 */
public class EntityStatsCommand extends Command {

    public EntityStatsCommand() {
        super("entitystats", null, "estats");
        setDescription("Print the statistics of the entity you're currently riding");
    }

    @Override
    public void call(String[] args) {

        if (mc.player.getRidingEntity() != null && mc.player.getRidingEntity() instanceof AbstractHorse) {
            AbstractHorse horse = (AbstractHorse) mc.player.getRidingEntity();
            float maxHealth = horse.getMaxHealth();
            double speed = MathUtils.INSTANCE.round(43.17 * horse.getAIMoveSpeed(), 2);
            double jump = MathUtils.INSTANCE.round(-0.1817584952 * Math.pow(horse.getHorseJumpStrength(), 3) + 3.689713992 * Math.pow(horse.getHorseJumpStrength(), 2) + 2.128599134 * horse.getHorseJumpStrength() - 0.343930367, 4);
            String ownerId = horse.getOwnerUniqueId() == null ? "Not tamed." : horse.getOwnerUniqueId().toString();

            PlayerProfile ownerProfile = UUIDManager.INSTANCE.getByString(ownerId);
            String ownerName = "";
            if (ownerProfile != null) ownerName = ownerProfile.getName();

            String builder = "&6Entity Statistics:" + "\n&cMax Health: " + maxHealth +
                    "\n&cSpeed: " + speed +
                    "\n&cJump: " + jump +
                    "\n&cOwner: " + ownerName;
            sendChatMessage(builder);
        } else if (mc.player.getRidingEntity() instanceof EntityLivingBase) {
            EntityLivingBase entity = (EntityLivingBase) mc.player.getRidingEntity();
            sendChatMessage("&6Entity Stats:\n&cMax Health: &b" + entity.getMaxHealth() + " &2HP" + "\n&cSpeed: &b" + MathUtils.INSTANCE.round(43.17 * entity.getAIMoveSpeed(), 2) + " &2m/s");
        } else {
            sendErrorMessage("&4&lError: &cNot riding a compatible entity.");
        }
    }
}

File: src/main/kotlin/com/lambda/client/command/commands/EntityStatsCommand.kt
package com.lambda.client.command.commands

import com.lambda.client.command.ClientCommand
import com.lambda.client.commons.utils.MathUtils
import com.lambda.client.manager.managers.UUIDManager
import com.lambda.client.util.text.MessageSendHelper
import net.minecraft.entity.EntityLivingBase
import net.minecraft.entity.passive.AbstractHorse
import kotlin.math.pow

object EntityStatsCommand : ClientCommand(
    name = "entitystats",
    alias = arrayOf("estats"),
    description = "Display statistics about the tamed, riding entity"
) {
    init {
        executeAsync {
            player?.ridingEntity?.let { entity ->
                if (entity !is EntityLivingBase) {
                    MessageSendHelper.sendErrorMessage("Not riding a compatible entity!")
                    return@executeAsync
                }

                val speed = MathUtils.round(43.17 * entity.aiMoveSpeed, 2)
                val maxHealth = entity.maxHealth
                val healthPercentage = MathUtils.round((entity.health / maxHealth) * 100, 2)

                if (entity is AbstractHorse) {
                    val jump = MathUtils.round(-0.1817584952
                        * entity.horseJumpStrength.pow(3.0) + 3.689713992
                        * entity.horseJumpStrength.pow(2.0) + 2.128599134
                        * entity.horseJumpStrength - 0.343930367, 4
                    )
                    val ownerId = entity.ownerUniqueId?.toString() ?: "Not tamed."
                    val ownerName = UUIDManager.getByString(ownerId)?.name ?: "Unknown owner."

                    MessageSendHelper.sendChatMessage("Entity Statistics:\n" +
                        "&cMax Health:&f $maxHealth\n" +
                        "&cHealth:&f $healthPercentage%\n" +
                        "&cSpeed:&f $speed\n" +
                        "&cJump:&f $jump\n" +
                        "&cOwner ID:&f $ownerId\n" +
                        "&cOwner Name:&f $ownerName"
                    )
                } else {
                    MessageSendHelper.sendChatMessage("Entity Statistics:\n" +
                        "&cMax Health:&f $maxHealth\n" +
                        "&cHealth:&f $healthPercentage%\n" +
                        "&cSpeed:&f $speed"
                    )
                }
            } ?: run {
                MessageSendHelper.sendErrorMessage("Not riding anything!")
            }

        }
    }
}